backend/src/test/java/com/github/nickklock/backend/controllers/SpotControllerTest.java

```diff
@@ -0,0 +1,46 @@
+package com.github.nickklock.backend.controllers;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
```

Comment:
I guess there's a good reason to use this configuration. Feel free to shine light unto us unknowing plebs.
Pro tip: You might as well document the explanation and add a comment explaining the why.

Reply:
tbh it's not needed to use @TestInstance right now. But for your knowledge 
`@TestInstance has two modes. One is LifeCycle.PER_METHOD (the default). The other is Lifecycle.PER_CLASS. The latter enables us to ask JUnit to create only one instance of the test class and reuse it between tests.`

@DirtiesContext cleans up everything after each method